test/markdown_renderers.js

```diff
@@ -0,0 +1,70 @@
+// import marked from 'marked';
+import { expect } from 'chai';
+import { customHeadingRenderer } from '../src/markdown_renderers';
+
+const marked = require('marked');
+
+describe('customHeadingRenderer', () => {
+    const renderer = new marked.Renderer();
+    renderer.heading = customHeadingRenderer;
+    marked.use({ renderer });
+
+    it('returns a header with ID param when ID is passed', () => {
+        const renderedTitle = function () {
+            return marked('# Welcome to Apify {#welcome-title-id}');
+        };
+        expect(renderedTitle()).to.equal(`
+            <h1>
+                <a 
+                    name="welcome-to-apify"
+                    href="#welcome-title-id"
+                    id="#welcome-title-id">
+                    <span class="header-link"></span>
+                </a>
+                Welcome to Apify
+            </h1>`);
+    });
+
+    it('adds a hashtag to ID if one isn\'t added', () => {
+        const renderedTitle = function () {
+            return marked('# Welcome to Apify {welcome-title-id}');
+        };
+        expect(renderedTitle()).to.equal(`
+            <h1>
+                <a 
+                    name="welcome-to-apify"
+                    href="#welcome-title-id"
+                    id="#welcome-title-id">
+                    <span class="header-link"></span>
+                </a>
+                Welcome to Apify
+            </h1>`);
+    });
+
+    it('returns a header without ID param when no ID is passed', () => {
```

Comment:
When there is no ID specified it should generate it from the heading text to be backward compatible